### PR TITLE
Add DuckDB Wasm example from repository

### PR DESCRIPTION
**Pull request by bluehat974**
Add DuckDB wasm examples (from duckdb-wasm repository)
https://github.com/duckdb/duckdb-wasm/blob/master/packages/duckdb-wasm/README.md

There is also more examples added (convert arrow table to JSON and export parquet)

### CODE REVIEW

**Comment by carlopi** [comment]
Thanks!
I think splitting the Wasm section in sub-sections is the way to go, and this seems a solid direction.
I would probably have wasm/overview.md be just a copy of old wasm.md, until that content do not find an alternative place.

The other pages probably would need a bit more text, at the very least on top, and potentially we could add some other sections (even later) like:
* how to embed duckdb-wasm, which files, how to serve them, example of deployments
* security would be nice to mention somewhere (basically that data processing happens on-device with no logging)

**Comment by bluehat974** [comment]
@carlopi 
I rollbacked to old wasm.md content and added small introduction for each pages in wasm folder.

For other ideas (embed duckdb-wasm or security), I need to think more and probably do it in another PR and do incremental develoment (there is so much knowledge I wish to copy/paste from discord or the unit test).

File: docs/api/wasm/extensions.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+layout: docu
+title: Extensions
+selected: Client APIs
+---
+
+Default [extensions](../../extensions/overview) enabled in DuckDB-wasm are HTTPFS and Parquet.
```
Comment (carlopi) [inline docs/api/wasm/extensions.md:7]: I think also json and fts (= full text search) are packaged by default.

File: docs/api/wasm/data_ingestion.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+---
+layout: docu
+title: Data Ingestion
+selected: Client APIs
+---
+
+DuckDB-wasm has multiple way to import data. 
+
+Basically, there are two steps to import data into DuckDB.
```
Comment (carlopi) [inline docs/api/wasm/data_ingestion.md:9]:
```suggestion
DuckDB-Wasm has multiple way to import data.

There are two steps to import data into DuckDB.
```

**Review by carlopi — APPROVED** [review]
Left another round with a few minor suggestions, but looks good to go after that.

I think on the overview.md it really needs a rework (as you suggested initially), but probably it makes more sense if I try to come up with a new structure and then we iterate starting from that.

Thanks a lot on this contribution, potentially if there are item that you think are worth inserting but don't have a place yet a possible way to keep track of them is opening an issue about DuckDB-Wasm documentation. Thanks again.

File: docs/api/wasm/extensions.md (hunk 1 above)
Comment (carlopi) [reply docs/api/wasm/extensions.md:7]: Actually, correcting myself, Parquet, FTS are alwasy packaged by default, while JSON and Excel are build-time opt-in. See https://github.com/duckdb/duckdb-wasm/blob/968ba722350d9be3988d7f44ac328c0ef0c8423a/lib/CMakeLists.txt#L238, I believe JSON and Excel

HTTPFS it's not really an extension, but a different re-implementation of it that comes bundled by default.

File: docs/api/wasm/overview.md [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+
+## Getting Started with DuckDB-Wasm
+
+A great starting point is to read the [DuckDB-Wasm launch blog post](https://duckdb.org/2021/10/29/duckdb-wasm.html)!
+
+Another great resource is the [Github repository](https://github.com/duckdb/duckdb-wasm).
+
+For details, see the full [DuckDB-Wasm API Documentation](https://shell.duckdb.org/docs/modules/index.html).
+
+_DuckDB-Wasm is fast! If you're here for performance numbers, head over to [our benchmarks](https://shell.duckdb.org/versus)._
```
[the remark this answers is not in the document]
Comment (bluehat974) [reply docs/api/wasm/overview.md:21]: I liked this url as it compare different browser database tools (sql.js, arquero...)

[the remark this answers is not in the document]
Comment (carlopi) [reply docs/api/wasm/overview.md:21]: Makes also sense. +1

[the remark this answers is not in the document]
Comment (bluehat974) [reply docs/api/wasm/overview.md]: Removed

File: docs/api/wasm/extensions.md (hunk 1 above)
Comment (bluehat974) [reply docs/api/wasm/extensions.md]: File changed

**Comment by Alex-Monahan** [comment]
Thank you for your contribution! This will be a real help to folks.

**Pull request merged**